### PR TITLE
Add valgrind = heavy to slower tests

### PR DESCRIPTION
**Pull request by loganharbour**
refs #18759

### CODE REVIEW

**Review by aeslaughter — DISMISSED** [review]

**Review by GiudGiud — DISMISSED** [review]
I dont think we should add this issue (to limit valgrind testing) to the issues of each test. 
Changes in our test harness are unrelated to what prompted the creation of these tests.
If we linked every test harness issue relevant for each test, we would have dozens already

**Comment by loganharbour** [comment]
In my opinion any change to a test that has a valid issue should be referenced.

**Comment by aeslaughter** [comment]
@loganharbour is correct all reasons for code, tests, documents, etc. for being changed should be listed.

**Comment by GiudGiud** [comment]
This is hardly a change to a test, you are moving it from one recipe to another. You are changing things with regards to runtime of the test suite, not wrt to testing of the code, design of the object. 
This issue is completely unrelated to the requirement fields of this test.

Besides, the reason for the test 'change' is listed in the commit, which can be found from a git blame on that line.

**Comment by loganharbour** [comment]
"hardly" a change is still a change. Such changes could result in something going from being tested to not being tested depending on what configurations are run. 

> Besides, the reason for the test 'change' is listed in the commit, which can be found from a git blame on that line.

This is true for most cases, but not all. Hence why an `issue` spec is needed in the first place.

**Comment by GiudGiud** [comment]
valgrind and valgrind heavy are both ran at the same time on a next merge. heavy is actually the lighter one and we wont be adding it to regular PR merges because it does not make sense. 'hardly' was just me being polite.

I stand my case that if we do this we should be listing way more issues regarding the test suite on each of these tests. Making this issue field useless because too long.

There are plenty of issues and PRs that change effectively nothing but do affect how these tests are run.

**Comment by aeslaughter** [comment]
@GiudGiud Git history is not enough for, NQA-1 auditors are looking for direct connection between a change and a reason. Inspecting logs, commits, and diffs is not good enough for them. Having a list of changes made in the test, on the website, is what they want. The information is obvious and accessible to non-coders. Also, it works w/o source code access which may be necessary for future applications.

**Comment by GiudGiud** [comment]
No change is made in the test. It s moved from one recipe to another.

Also under this logic you need to add to every test the issue for the creation of recover -p 2 testing. This was a change that affected every single of these tests. They not only got moved around in recipes, but they were tested in a different configuration. Much much bigger deal than this PR actually.

I dont deny this system is great and well conceived. I'm pushing back on linking trivial unrelated 'changes' to tests that have a specific purpose.

**Comment by lindsayad** [comment]
My opinion is that for actual users and developers of our code adding a ticket number for every trivial test spec change obfuscates the purpose of the test. You may have developers interested in learning about the substance of the test clicking through many tickets that really have no meaning on its substance. 

But if the bureaucracy of NQA-1 requires this, then we must do it I guess even though its a detriment to actual use of our code base in my opinion.

**Comment by GiudGiud** [comment]
What does our PLN-4005 say? I cant see issues in the RTM section, I must be looking at the wrong one

**Comment by aeslaughter** [comment]
The issues are linked on the RTM: https://mooseframework.inl.gov/sqa/moose_rtm.html

PLN-4005 states: 
```
Reference at least one of the following: 
1.	An issue that motivated the associated test case.
2.	The CR that implemented the test case.
3.	The revision when it was created.
```
I will change my response and say it is not necessary to add it. Personally, I think it is better to add the issue, but if you don't want to I can respect that and will not mandate it. 😉

File: modules/combined/test/tests/additive_manufacturing/tests [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [Tests]
-  issues = '#15795'
+  issues = '#15795 #18970'
```
Comment (loganharbour) [inline modules/combined/test/tests/additive_manufacturing/tests:2]:
```suggestion
  issues = '#15795'
```

File: modules/contact/test/tests/normal-nodal-lm-tan-tolerance/tests [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [Tests]
   design = 'modules/contact/index.md'
-  issues = '#10950'
+  issues = '#10950 #18970'
```
Comment (loganharbour) [inline modules/contact/test/tests/normal-nodal-lm-tan-tolerance/tests:3]:
```suggestion
  issues = '#10950'
```

File: modules/level_set/test/tests/kernels/olsson_reinitialization/tests [hunk 3]
```diff
@@ -1,10 +1,11 @@
 [Tests]
-  issues = '#8465'
+  issues = '#8465 #18970'
```
Comment (loganharbour) [inline modules/level_set/test/tests/kernels/olsson_reinitialization/tests:2]:
```suggestion
  issues = '#8465'
```

File: modules/level_set/test/tests/reinitialization/tests [hunk 4]
```diff
@@ -7,8 +7,9 @@
     exodiff = master_original.e
     cli_args = 'Executioner/num_steps=2 Outputs/file_base=master_original'
     method = '!dbg' # complete reinitilization solve, debug is slow and not needed
+    valgrind = HEAVY
     requirement = "The level set module shall be capable of solving the level set equation with original reinitialization."
-    issues = '#14849'
+    issues = '#14849 #18970'
```
Comment (loganharbour) [inline modules/level_set/test/tests/reinitialization/tests:12]:
```suggestion
    issues = '#14849'
```

File: modules/level_set/test/tests/reinitialization/tests [hunk 5]
```diff
@@ -17,7 +18,8 @@
     exodiff = master_modified.e
     cli_args = 'Executioner/num_steps=2 MultiApps/reinit/input_files=reinit_modified.i Outputs/file_base=master_modified'
     method = '!dbg' # complete reinitilization solve, debug is slow and not needed
+    valgrind = HEAVY
     requirement = "The level set module shall be capable of solving the level set equation with modified reinitialization."
-    issues = '#8465'
+    issues = '#8465 #18970'
```
Comment (loganharbour) [inline modules/level_set/test/tests/reinitialization/tests:23]:
```suggestion
    issues = '#8465'
```

File: modules/level_set/test/tests/verification/1d_level_set_mms/tests [hunk 6]
```diff
@@ -1,5 +1,5 @@
 [Tests]
-  issues = '#8465'
+  issues = '#8465 #18970'
```
Comment (loganharbour) [inline modules/level_set/test/tests/verification/1d_level_set_mms/tests:2]:
```suggestion
  issues = '#8465'
```

File: modules/navier_stokes/test/tests/finite_volume/cns/natural_convection/tests [hunk 7]
```diff
@@ -1,5 +1,5 @@
 [Tests]
-  issues = '#16758'
+  issues = '#16758 #18970'
```
Comment (loganharbour) [inline modules/navier_stokes/test/tests/finite_volume/cns/natural_convection/tests:2]:
```suggestion
  issues = '#16758'
```

File: modules/navier_stokes/test/tests/finite_volume/cns/pressure_outlet/tests [hunk 8]
```diff
@@ -1,5 +1,5 @@
 [Tests]
-  issues = '#16758'
+  issues = '#16758 #18970'
```
Comment (loganharbour) [inline modules/navier_stokes/test/tests/finite_volume/cns/pressure_outlet/tests:2]:
```suggestion
  issues = '#16758'
```

File: modules/navier_stokes/test/tests/finite_volume/cns/straight_channel_porosity_step/tests [hunk 9]
```diff
@@ -1,5 +1,5 @@
 [Tests]
-  issues = '#16758'
+  issues = '#16758 #18970'
```
Comment (loganharbour) [inline modules/navier_stokes/test/tests/finite_volume/cns/straight_channel_porosity_step/tests:2]:
```suggestion
  issues = '#16758'
```

File: modules/navier_stokes/test/tests/finite_volume/fvbcs/wall_function/tests [hunk 10]
```diff
@@ -1,5 +1,5 @@
 [Tests]
-  issues = '#18273'
+  issues = '#18273 #18970'
```
Comment (loganharbour) [inline modules/navier_stokes/test/tests/finite_volume/fvbcs/wall_function/tests:2]:
```suggestion
  issues = '#18273'
```

File: modules/navier_stokes/test/tests/finite_volume/ins/channel-flow/tests [hunk 11]
```diff
@@ -35,27 +35,30 @@
     requirement = 'The system shall be able to transport arbitrary scalar field variables in a fluid flow field.'
     ad_indexing_type = 'global'
     design = 'INSFVScalarFieldAdvection.md'
-    issues = '#16732'
+    issues = '#16732 #18970'
```
Comment (loganharbour) [inline modules/navier_stokes/test/tests/finite_volume/ins/channel-flow/tests:38]:
```suggestion
    issues = '#16732'
```

File: modules/navier_stokes/test/tests/finite_volume/ins/channel-flow/tests [hunk 12]
```diff
@@ -35,27 +35,30 @@
...
+    valgrind = HEAVY
   []
   [momentum-outflow-bcs]
     type = 'Exodiff'
     input = 2d-rc-no-slip-outflow-bcs.i
     exodiff = 2d-rc-no-slip-outflow-bcs_out.e
     requirement = 'The system shall be able to use flux boundary conditions for the momentum and match results produced by using flux kernels.'
     design = 'INSFVMomentumAdvectionOutflowBC.md'
-    issues = '#16854'
+    issues = '#16854 #18970'
```
Comment (loganharbour) [inline modules/navier_stokes/test/tests/finite_volume/ins/channel-flow/tests:47]:
```suggestion
    issues = '#16854'
```

File: modules/navier_stokes/test/tests/finite_volume/ins/channel-flow/tests [hunk 13]
```diff
@@ -35,27 +35,30 @@
...
+    valgrind = HEAVY
   []
   [extrapolated-outlet-pressure]
     type = 'Exodiff'
     input = 2d-rc-no-slip-extrapolated-outlet-pressure.i
     exodiff = 2d-rc-no-slip-extrapolated-outlet-pressure_out.e
     requirement = 'The system shall be able to extrapolate a pressure value at a fully developed outflow boundary and use a mean pressure approach to eliminate the nullspace for the pressure.'
     design = 'INSFVMassAdvectionOutflowBC.md'
-    issues = '#16854'
+    issues = '#16854 #18970'
```
Comment (loganharbour) [inline modules/navier_stokes/test/tests/finite_volume/ins/channel-flow/tests:58]:
```suggestion
    issues = '#16854'
```

File: modules/navier_stokes/test/tests/finite_volume/ins/lid-driven/tests [hunk 14]
```diff
@@ -1,6 +1,6 @@
 [Tests]
   design = 'navier_stokes/index.md'
-  issues = '#15640'
+  issues = '#15640 #18970'
```
Comment (loganharbour) [inline modules/navier_stokes/test/tests/finite_volume/ins/lid-driven/tests:3]:
```suggestion
  issues = '#15640'
```

File: modules/navier_stokes/test/tests/finite_volume/ins/mixing_length_eddy_viscosity_aux/tests [hunk 15]
```diff
@@ -1,5 +1,5 @@
 [Tests]
-  issues = '#18666'
+  issues = '#18666 #18970'
```
Comment (loganharbour) [inline modules/navier_stokes/test/tests/finite_volume/ins/mixing_length_eddy_viscosity_aux/tests:2]:
```suggestion
  issues = '#18666'
```

File: modules/navier_stokes/test/tests/finite_volume/ins/mixing_length_total_viscosity_material/tests [hunk 16]
```diff
@@ -1,5 +1,5 @@
 [Tests]
-  issues = '#18666'
+  issues = '#18666 #18970'
```
Comment (loganharbour) [inline modules/navier_stokes/test/tests/finite_volume/ins/mixing_length_total_viscosity_material/tests:2]:
```suggestion
  issues = '#18666'
```

File: modules/navier_stokes/test/tests/finite_volume/pins/channel-flow/heated/tests [hunk 17]
```diff
@@ -1,6 +1,6 @@
 [Tests]
   [heated]
-    issues = '#16756'
+    issues = '#16756 #18970'
```
Comment (loganharbour) [inline modules/navier_stokes/test/tests/finite_volume/pins/channel-flow/heated/tests:3]:
```suggestion
    issues = '#16756'
```

File: modules/navier_stokes/test/tests/finite_volume/pins/channel-flow/heated/tests [hunk 18]
```diff
@@ -40,9 +43,10 @@
     exodiff = rc-heated-solid-fluid.e
     method = "!dbg"
     abs_zero = 1e-9
-    issues = '#16756'
+    issues = '#16756 #18970'
```
Comment (loganharbour) [inline modules/navier_stokes/test/tests/finite_volume/pins/channel-flow/heated/tests:46]:
```suggestion
    issues = '#16756'
```

File: modules/navier_stokes/test/tests/finite_volume/pins/channel-flow/heated/tests [hunk 19]
```diff
@@ -51,9 +55,10 @@
     input = 2d-rc-heated-boussinesq.i
     exodiff = rc-heated-boussinesq.e
     method = "!dbg"
-    issues = '#16756'
+    issues = '#16756 #18970'
```
Comment (loganharbour) [inline modules/navier_stokes/test/tests/finite_volume/pins/channel-flow/heated/tests:58]:
```suggestion
    issues = '#16756'
```

File: modules/navier_stokes/test/tests/ins/boussinesq/tests [hunk 20]
```diff
@@ -1,11 +1,12 @@
 [Tests]
   design = 'INSADBoussinesqBodyForce.md'
-  issues = '#15099'
+  issues = '#15099 #18970'
```
Comment (loganharbour) [inline modules/navier_stokes/test/tests/ins/boussinesq/tests:3]:
```suggestion
  issues = '#15099'
```

File: modules/phase_field/test/tests/ad_coupled_gradient_dot/tests [hunk 21]
```diff
@@ -1,12 +1,13 @@
 [Tests]
   design = 'Coupleable.md'
-  issues = '#16167'
+  issues = '#16167 #18970'
```
Comment (loganharbour) [inline modules/phase_field/test/tests/ad_coupled_gradient_dot/tests:3]:
```suggestion
  issues = '#16167'
```

File: modules/phase_field/test/tests/initial_conditions/tests [hunk 22]
```diff
@@ -1,5 +1,5 @@
 [Tests]
-  issues = '#6858 #5816 #7358 #8721 #12000 #6655 #9174'
+  issues = '#6858 #5816 #7358 #8721 #12000 #6655 #9174 #18970'
```
Comment (loganharbour) [inline modules/phase_field/test/tests/initial_conditions/tests:2]:
```suggestion
  issues = '#6858 #5816 #7358 #8721 #12000 #6655 #9174'
```

File: modules/stochastic_tools/examples/sobol/tests [hunk 23]
```diff
@@ -1,6 +1,6 @@
 [Tests]
   design = sobol.md
-  issues = '#14933 #15558'
+  issues = '#14933 #15558 #18970'
```
Comment (loganharbour) [inline modules/stochastic_tools/examples/sobol/tests:3]:
```suggestion
  issues = '#14933 #15558'
```

File: modules/tensor_mechanics/test/tests/ad_plastic/tests [hunk 24]
```diff
@@ -1,5 +1,5 @@
 [Tests]
-  issues = '#12650'
+  issues = '#12650 #18970'
```
Comment (loganharbour) [inline modules/tensor_mechanics/test/tests/ad_plastic/tests:2]:
```suggestion
  issues = '#12650'
```

File: modules/tensor_mechanics/test/tests/ad_smeared_cracking/tests [hunk 25]
```diff
@@ -1,6 +1,6 @@
 [Tests]
 design = 'source/materials/ADComputeSmearedCrackingStress.md'
-issues = '#6815 #9227 #15311'
+issues = '#6815 #9227 #15311 #18970'
```
Comment (loganharbour) [inline modules/tensor_mechanics/test/tests/ad_smeared_cracking/tests:3]:
```suggestion
issues = '#6815 #9227 #15311'
```

File: modules/tensor_mechanics/test/tests/combined_creep_plasticity/tests [hunk 26]
```diff
@@ -1,12 +1,13 @@
 [Tests]
-  issues = '#6873 #8651'
+  issues = '#6873 #8651 #18970'
```
Comment (loganharbour) [inline modules/tensor_mechanics/test/tests/combined_creep_plasticity/tests:2]:
```suggestion
  issues = '#6873 #8651'
```

File: modules/tensor_mechanics/test/tests/crystal_plasticity/stress_update_material_based/tests [hunk 27]
```diff
@@ -74,7 +74,8 @@
     input = 'use_substep_dt.i'
     csvdiff = 'use_substep_dt_out.csv'
     allow_warnings = true
-    issues = '#17340'
+    valgrind = HEAVY
+    issues = '#17340 #18970'
```
Comment (loganharbour) [inline modules/tensor_mechanics/test/tests/crystal_plasticity/stress_update_material_based/tests:78]:
```suggestion
    issues = '#17340'
```

File: modules/tensor_mechanics/test/tests/crystal_plasticity/user_object_based/tests [hunk 28]
```diff
@@ -105,7 +105,8 @@
     input = 'use_substep_dt.i'
     csvdiff = 'use_substep_dt_out.csv'
     allow_warnings = true
-    issues = '#17340'
+    valgrind = HEAVY
+    issues = '#17340 #18970'
```
Comment (loganharbour) [inline modules/tensor_mechanics/test/tests/crystal_plasticity/user_object_based/tests:109]:
```suggestion
    issues = '#17340'
```

File: modules/tensor_mechanics/test/tests/multi_power_law/tests [hunk 29]
```diff
@@ -1,10 +1,11 @@
 [Tests]
-  issues = '#18850'
+  issues = '#18850 #18970'
```
Comment (loganharbour) [inline modules/tensor_mechanics/test/tests/multi_power_law/tests:2]:
```suggestion
  issues = '#18850'
```

File: modules/tensor_mechanics/test/tests/substepping/tests [hunk 30]
```diff
@@ -1,5 +1,5 @@
 [Tests]
-  issues = '#14757'
+  issues = '#14757 #18970'
```
Comment (loganharbour) [inline modules/tensor_mechanics/test/tests/substepping/tests:2]:
```suggestion
  issues = '#14757'
```

**Review by loganharbour — COMMENTED** [review]
Stubbornness is not worth my time

**Comment by GiudGiud** [comment]
Thanks! Your next beer's on me

**Review by aeslaughter — APPROVED** [review]

**Pull request merged**